### ISSUE TITLE
Any ways to create BevelFilter?

### ISSUE THREAD

**Issue by CrazyFlasher** [Gamua/Starling-Framework#1018]
Are there any ready implementations of BevelFilter?
Something like in native flash display list?
![gif](https://user-images.githubusercontent.com/1607138/35757398-7ef1b304-0877-11e8-932c-6b8ca492d8a3.gif)
![32aaacf545](https://user-images.githubusercontent.com/1607138/35757425-99428954-0877-11e8-8345-a92246f05ddc.png)

**Comment by varadig** [comment]
http://wiki.starling-framework.org/extensions/dynamic_lighting

**Comment by CrazyFlasher** [comment]
Yes, I know about dynamic lightning. But I have kinematic animation (not sprite sheet) with starling display object tree created with Flump. I cannot create normal maps in my case

**Comment by CrazyFlasher** [comment]
I really don't want do dig into AGAL to write custom shader, so any help will be appreciated :)

**Comment by devon-o** [comment]
That makes sense. I'll drop an email later (may be tomorrow) for more details.

**Comment by oskaraqp** [comment]
Just as a suggestion your kinematic animation works moving images(textures extracted from your spritesheet) around so probably the dynamic_lightning approach would work if you can modify Flump in order to use the normal maps and overlap it with the texture(copying the moving and applying it), also a workaround could be to copy the sprite currently being using into a bitmap and apply the filter, time ago I was trying to get the bevel filter to work on Starling but there were some issues to replicate as it was needed a bitmap anyway in order to duplicate the borders.

**Comment by CrazyFlasher** [comment]
If I apply filter to each body part used in animation, I'll get bad result:
Expected result:
![ex](https://user-images.githubusercontent.com/1607138/35793999-91ccc87e-0a5c-11e8-91d6-9c3c326f6dfc.png)

Actual result:
![ac](https://user-images.githubusercontent.com/1607138/35794004-951ee05c-0a5c-11e8-862c-4207b39c480e.png)

**Comment by CrazyFlasher** [comment]
Btw, inner glow can be used instead of bevel in my case. But I think the problem to implement inner glow is the same as bevel?

**Comment by PrimaryFeather** [comment]
Mhm, with the existing filters, only the Dynamic Lighting extension will give you a bevel-like result. However, it is rendered per mesh, thus the result you're seeing is different than what you'd get with a filter.

It would be possible to create an inner glow filter — with a few adaptations of the Starling code. The _DropShadowFilter_ and the _GlowFilter_ actually work in two steps:

1. Blur the original object (using a _BlurFilter) and tint the result with a single color.
2. Combine blurred object with original object using a _CompositeFilter_.

Right now, the _CompositeFilter_ can only draw textures on top of each other. But it was always designed to get a "knockout" mode, where the alpha of one layer is inverted. With that implemented, it would work.

However, there's still the performance penalty, of course: a blur filter takes several passes, and then it needs to be combined with the original texture. If that makes sense for you depends on the number of objects you need to process like this, and the devices you want to target.

**Comment by hardcoremore** [comment]
Yes, Bevel filter would be great addition to Starling!
